### PR TITLE
test: Suppress epoll_ctl data race

### PR DESCRIPTION
**Pull request by maflcko**
Happens intermittently: https://cirrus-ci.com/task/5462892373868544?command=ci#L5385

### CODE REVIEW

**Comment by hebasto** [comment]
I think our code should fixed to prevent data race in `MaybeCompactWalletDB` instead of silencing a warning as it seems related to `CScheduler` rather internal BDB code.

**Comment by MarcoFalke** [comment]
@hebasto I agree, but until someone figured out where the issue happens and how to fix it, we should suppress the intermittent failures, because they are hurting the project.

File: test/sanitizer_suppressions/tsan [hunk 1]
```diff
@@ -20,6 +20,7 @@ race:fHaveGenesis
 race:ProcessNewBlock
 race:ThreadImport
 race:LoadWallet
+race:MaybeCompactWalletDB
```
Comment (hebasto) [inline test/sanitizer_suppressions/tsan:23]:
```suggestion
race:epoll_ctl
```

as log contains
```
SUMMARY: ThreadSanitizer: data race (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/bitcoind+0x8e332) in epoll_ctl
```
?

**Review by hebasto — COMMENTED** [review]
Approach ACK fa5c8f4b7027200d8f54aad83ab5c1bfc5a42c90.

It seems the data race in [`epoll_ctl`](https://linux.die.net/man/2/epoll_ctl) could be false positive, and be caused by clang tsan instrumentation code (e.g., https://lists.llvm.org/pipermail/llvm-commits/Week-of-Mon-20131007/190642.html). But not for sure, of course.

Also there is a possibility that this issue is/will be fixed in newer `libevent` versions.

File: test/sanitizer_suppressions/tsan (hunk 1 above)
Comment (MarcoFalke) [reply test/sanitizer_suppressions/tsan]: thanks, fixed

**Review by hebasto — APPROVED** [review]
ACK fa949b3c1325693ea7ecc5556b2de50d2a6c9ead, I have reviewed the code and it looks OK, I agree it can be merged.

**Comment by MarcoFalke** [comment]
Not sure why, but this doesn't work:

https://cirrus-ci.com/task/5734386425397248?command=ci#L5451

**Comment by hebasto** [comment]
> Not sure why, but this doesn't work:
> 
> https://cirrus-ci.com/task/5734386425397248?command=ci#L5451

Indeed (

**Comment by hebasto** [comment]
@MarcoFalke 
> Not sure why, but this doesn't work:
> 
> https://cirrus-ci.com/task/5734386425397248?command=ci#L5451

See #20745.

**Pull request merged**